js/ui/windowManager.js

```diff
@@ -2343,6 +2365,23 @@ var WindowManager = new Lang.Class({
             return;
         }
 
+        if (actor._animatableSurface) {
+            // This is an event that may cause an animation
+            // on the window.
+            let attachedEffect = actor._animatableSurface.highest_priority_attached_effect_for_event('unminimize');
+
+            if (attachedEffect) {
+                // Guaranteed to work, since the minimize effect is
+                // only available if this window has an icon
+                let icon = getTaskbarIconGeometryForWindow(actor.meta_window);
+                attachedEffect.activate('unminimize', {
+                    icon: icon,
+                    complete: () => shellwm.completed_unminimize(actor)
+                });
```

Comment:
This code is similar to the `minimize` version above, so maybe it'd be good to abstract the blocks in a function. The function can already check whether the `attachedEffect` is `null` and return a boolean, so we know whether to return here or not.

Reply:
Yup, created a new helper in a separate commit.